Scripts/Runtime/Entities/Tasks/AbstractTaskDriverSystem.cs

```diff
@@ -43,6 +47,9 @@ protected AbstractTaskDriverSystem()
             m_TaskDataLookup = new VirtualDataLookup();
             m_TaskDrivers = new List<AbstractTaskDriver>();
             m_UpdateJobData = new List<JobTaskWorkConfig>();
+            m_TaskDriverIDProvider = new IDProvider();
+            //Reserves the first ID of 1 for System Level -> See SYSTEM_LEVEL_CONTEXT
+            m_TaskDriverIDProvider.GetNextID();
```

Comment:
This makes an assumption about how `IDProvider` works which matches behaviour today but not necessarily in the future. Suggest you store the system reserved ID and use it rather than a constant.

If it's really critical then at least assert that the value is what you're assuming it is.

Reply:
Good point, fixed.